SimulationIOFilters/ImportDelamData.cpp

```diff
@@ -0,0 +1,568 @@
+    setErrorCondition(-2014, ss);
+    return;
+  }
+
+  bvidStdOutFile.close();
+}
+
+// -----------------------------------------------------------------------------
+size_t ImportDelamData::getBvidFileLineCount()
+{
+  QFile bvidFile(m_CSDGMFile);
+  if(!bvidFile.open(QFile::ReadOnly))
+  {
+    QString ss = QObject::tr("Cannot open Bvid File for reading.");
+    setErrorCondition(-2015, ss);
+    return 0;
+  }
+
+  size_t lineCount = 0;
+  while(!bvidFile.readLine().isEmpty())
+  {
+    lineCount++;
+  }
+
+  bvidFile.close();
+
+  return lineCount;
+}
+
+// -----------------------------------------------------------------------------
+void ImportDelamData::execute()
+{
+  initialize();
+  dataCheck();
+  if(getErrorCode() < 0)
+  {
+    return;
+  }
+
+  readCSDGMFile();
+}
+
+// -----------------------------------------------------------------------------
+void ImportDelamData::readCSDGMFile()
+{
+  QFile csdgmFile(m_CSDGMFile);
+  if(!csdgmFile.open(QFile::ReadOnly))
+  {
+    QString ss = QObject::tr("Cannot open Bvid File for reading.");
+    setErrorCondition(-2016, ss);
+    return;
+  }
+
+  QString csdgmLine = csdgmFile.readLine();
+  size_t lineCount = 1;
+  while(!csdgmLine.isEmpty())
+  {
+    QStringList tokens = csdgmLine.split(' ', QString::SplitBehavior::SkipEmptyParts);
+    if(tokens.size() != 4)
+    {
+      std::stringstream ss;
+      ss << "CSDGM File line " << lineCount << ": line does not have 4 values.";
+      setErrorCondition(-2017, QString::fromStdString(ss.str()));
+      return;
+    }
+
+    bool ok = false;
+    float xCoord = tokens[0].toFloat(&ok);
+    if(!ok)
+    {
+      std::stringstream ss;
+      ss << "CSDGM File line " << lineCount << ": Could not convert X value to a float.";
+      setErrorCondition(-2018, QString::fromStdString(ss.str()));
+      return;
+    }
+
+    float yCoord = tokens[1].toFloat(&ok);
+    if(!ok)
+    {
+      std::stringstream ss;
+      ss << "CSDGM File line " << lineCount << ": Could not convert Y value to a float.";
+      setErrorCondition(-2019, QString::fromStdString(ss.str()));
+      return;
+    }
+
+    float zCoord = tokens[2].toFloat(&ok);
+    if(!ok)
+    {
+      std::stringstream ss;
+      ss << "CSDGM File line " << lineCount << ": Could not convert Z value to a float.";
+      setErrorCondition(-2020, QString::fromStdString(ss.str()));
+      return;
+    }
+
+    float value = tokens[3].toFloat(&ok);
+    if(!ok)
+    {
+      std::stringstream ss;
+      ss << "CSDGM File line " << lineCount << ": Could not convert 4th value to a float.";
+      setErrorCondition(-2021, QString::fromStdString(ss.str()));
+      return;
+    }
+
+    DataContainer& dc = *getDataContainerArray()->getDataContainer(getDataContainerName());
+    AttributeMatrix& am = *dc.getAttributeMatrix(getCellAttributeMatrixName());
+    FloatArrayType& da = *am.getAttributeArrayAs<FloatArrayType>(getDataArrayName());
```

Comment:
It doesn't look like you are checking for null before converting a shared_ptr to a reference. Null references are not a thing. And then you are using a possibly null reference in the next line without checking if the **pointer** is null.